### PR TITLE
Fixes for NXP SE050 ECC create and key store id

### PR DESCRIPTION
**Pull request by dgarske**
Fix for ECC create key public export size and key size bits. Fix for key store ID vs key ID.

### CODE REVIEW

File: wolfcrypt/src/port/nxp/se050_port.c [hunk 1]
```diff
@@ -378,56 +391,107 @@ void se050_aes_free(Aes* aes)
...
     if (cfg_se050_i2c_pi == NULL) {
         return WC_HW_E;
     }
+    if (key->keyId <= 0) {
+        return BAD_FUNC_ARG;
+    }
+
+    keySize = key->dp->size;
+    keySizeBits = keySize * 8;
+    if (keySizeBits > SSS_MAX_ECC_BITS)
```
Comment (SparkiDev) [inline wolfcrypt/src/port/nxp/se050_port.c:489]: This looks odd.

File: wolfcrypt/src/ecc.c [hunk 2]
```diff
@@ -7098,7 +7091,9 @@ int wc_ecc_verify_hash_ex(mp_int *r, mp_int *s, const byte* hash,
     }
 #endif
 
-#if defined(WOLFSSL_ATECC508A) || defined(WOLFSSL_ATECC608A)
+#if defined(WOLFSSL_ATECC508A) || defined(WOLFSSL_ATECC608A) || \
+    defined(WOLFSSL_CRYPTOCELL) || defined(WOLFSSL_SILABS_SE_ACCEL) || \
+    defined(WOLFSSL_KCAPI_ECC) || defined(WOLFSSL_SE050)
     /* Extract R and S */
     err = mp_to_unsigned_bin(r, &sigRS[0]);
```
Comment (SparkiDev) [inline wolfcrypt/src/ecc.c:7098]: Is this front padding with zeros?

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: wolfcrypt/src/port/nxp/se050_port.c (hunk 1 above)
Comment (dgarske) [reply wolfcrypt/src/port/nxp/se050_port.c:489]: This is my hack for handling the 521 bit curve. I will find a better solution.

File: wolfcrypt/src/ecc.c (hunk 2 above)
Comment (SparkiDev) [reply wolfcrypt/src/ecc.c]: Thanks!

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**